zaza/openstack/charm_tests/hacluster/tests.py

```diff
@@ -0,0 +1,104 @@
+        super(HaclusterTest, cls).setUpClass()
+        cls.vip = os.environ.get("TEST_VIP00")
+
+    def test_900_action_cleanup(self):
+        """The services can be cleaned up. """
+        status = zaza.model.get_status().applications[self.application_name]
+
+        # libjuju juju status no longer has units for subordinate charms
+        # Use the application it is subordinate-to to check workload status
+        if status.get("units") is None and status.get("subordinate-to"):
+            primary_status = juju_utils.get_application_status( \
+                    status.get("subordinate-to")[0])
+            leader = None
+            for unit in primary_status["units"]:
+                if primary_status["units"][unit].get('leader'):
+                    leader = unit
+        leader_unit = zaza.model.get_unit_from_name(leader)
+        assert leader_unit.workload_status == "active"
+
+        if primary_status["units"][leader].get("subordinates"):
+            for subordinate in primary_status["units"][leader]["subordinates"]:
+                logging.info("Cleaning {}".format(subordinate))
+                __action = "cleanup"
+                action_id = zaza.model.run_action(subordinate, "cleanup")
+                zaza.model.block_until_unit_wl_status(leader, "active")
+                assert "success" in action_id.data["results"]["result"], (
+                        "Set hacluster action {} failed: {}"
+                        .format(__action, action_id.data))
+
+                logging.info("Cleaning action w/resource {}"
+                        .format(subordinate))
+                params = {'resource': 'res_ks_haproxy'}
+                __action = "cleanup res_ks_haproxy"
+                zaza.model.run_action(subordinate, "cleanup", \
+                        action_params=params)
+                assert "success" in action_id.data["results"]["result"], (
+                    "Set hacluster action {} failed: {}"
+                    .format(__action, action_id.data))
+                zaza.model.block_until_unit_wl_status(leader, "active")
+
+    def test_910_pause_and_resume(self):
+        """The services can be paused and resumed."""
+        logging.debug('Checking pause and resume actions...')
+
+        status = zaza.model.get_status().applications[self.application_name]
+
+        # libjuju juju status no longer has units for subordinate charms
+        # Use the application it is subordinate-to to check workload status
+        if status.get("units") is None and status.get("subordinate-to"):
+            primary_status = juju_utils.get_application_status(
+                    status.get("subordinate-to")[0])
+            leader = None
+            for unit in primary_status["units"]:
+                if primary_status["units"][unit].get('leader'):
+                    leader = unit
+        leader_unit = zaza.model.get_unit_from_name(leader)
+        assert leader_unit.workload_status == "active"
```

Comment:
Like above: We don't need this check. And in fact, you can remove the leader_unit as we never use it for anything else.